modules/storages/app/common/storages/adapters/providers/share_point/queries/file_info_query.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module Adapters
+    module Providers
+      module SharePoint
+        module Queries
+          class FileInfoQuery < Base
+            FIELDS = %w[id name fileSystemInfo file folder size createdBy lastModifiedBy parentReference webUrl].freeze
+
+            def call(auth_strategy:, input_data:)
+              identifier_hash = split_identifier(input_data.file_id)
+              drive_id = identifier_hash[:drive_id]
+              item_id = identifier_hash[:location]
```

Comment:
Mh, so I get that we define a `split_identifier` method (probably in `Base`) and expect it to return a very specific hash that at least contains the keys `:drive_id` and `:location` and if those keys were missing, I'd expect this to be a programming error.

Thus I was first inclined to ask for `identifier_hash.fetch(...)`, but then I realized that you probably built the perfect code for Ruby 3's destructuring of hashes, didn't you?

What do you think about this?

```ruby
split_identifier(input_data.file_id) => { drive_id:, location: item_id }
```

Of course it would be a bit nicer if we didn't have to rename `location` to `item_id`, so I'd also be happy with

```ruby
split_identifier(input_data.file_id) => { drive_id:, location: }
```

which probably reads a bit nicer.